lib/vrl/compiler/src/expression/assignment.rs

```diff
@@ -296,9 +296,7 @@ impl Target {
             }
 
             External(path) => {
-                let _ = ctx
-                    .target_mut()
-                    .target_insert(path.as_ref().unwrap_or(&LookupBuf::root()), value);
```

Comment:
I did notice that we used `unwrap_or` here, instead of `unwrap_or_else`, which might have already solved the performance issue.

It's interesting that Clippy didn't ding this, but I guess it's because we take a reference, so it assumes it's cheap?

Either way, I believe the refactor to make the code easier to understand.